elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceTable.java

```diff
@@ -134,12 +157,39 @@ public void resolveAndStoreAllReferencesAndJoins(Queryable queryable) {
             resolvedJoinProjections.put(key, new HashMap<>());
         }
 
-        FormulaValidator validator = new FormulaValidator(metaDataStore);
+        if (!globalTablesContext.containsKey(key)) {
+
+            if (!isNested) {
+                Type<?> entityClass = dictionary.getEntityClass(key.getName(), key.getVersion());
+                SQLTable table = (SQLTable) metaDataStore.getTable(entityClass);
+
+                TableContext tableCtx = new TableContext(key.getAlias(),
+                                                         dialect,
+                                                         prepareColumnArgsMap(table.getArguments(), TBL_PREFIX));
+                globalTablesContext.put(key, tableCtx);
+                Set<Type<?>> visitedTableContext = new HashSet<>();
+                populateTableContext(entityClass, tableCtx, StringUtils.EMPTY, visitedTableContext);
+            } else {
+                TableContext tableCtx = new TableContext(key.getAlias(),
+                                                         dialect,
+                                                         Collections.emptyMap());
```

Comment:
With nesting, arguments maybe required in either the inner or outer queries.  We'll want to preserve them so they are available for substitution at all levels.